src/coreComponents/mesh/generators/WellBlock.hpp

```diff
@@ -0,0 +1,297 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2020-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef GEOSX_WELLBLOCK_HPP
+#define GEOSX_WELLBLOCK_HPP
+
+#include "mesh/generators/WellBlockABC.hpp"
+#include "mesh/generators/InternalWellGenerator.hpp"
+
+
+namespace geosx
+{
+
+/**
+ * Implementation of the WellBlock responsible for modification/creation capabilities.
+ */
+class WellBlock : public WellBlockABC
+{
+public:
+  /**
+   * @brief Constructor
+   * @param internallWellGenerator
+   */
+  WellBlock( const InternalWellGenerator & internalWellGenrator );
+
+
+  /**
+   * @name Getters / Setters
+   */
+  ///@{
+
+  // getters for element data
+
+  /**
+   * @brief Get the global number of well elements.
+   * @return the global number of elements
+   */
+  globalIndex getNumElements() const { return m_numElems; }
```

Comment:
Please add the `override` keyword when you override, it's very useful to prevent spelling or refactoring mistakes.
Plus we usually use the `numElements` naming convention. (see the `CellBlockManagerABC`, `FaceBlockABC` and `CellBlockABC` classes.)

Reply:
I'll add it, I copied getters from InternalWellGenerator, they don't follow tje convention, I'll change that in WellBlock at least.

Reply:
> I'll add it, I copied getters from InternalWellGenerator, they don't follow tje convention, I'll change that in WellBlock at least.

Oh, I did not know that `InternalWellGenerator` did not respect the conventions... We need to fix this as well 🤣